### PR TITLE
build.ps1: Consolidate the redist installation mimicking Program Files

### PR DESCRIPTION
**Pull request by tristanlabelle**
Updates the consolidation step to write to folders mimicking Program Files native and WoW directories.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -695,8 +692,30 @@ function Consolidate-RedistInstall($Arch)
...
+  }
+  elseif ($Arch -eq $ArchX86)
+  {
+    $ProgramFilesName = "Program Files (x86)"
+  }
+  elseif (($HostArch -eq $HostArm64) -and ($Arch -eq $ArchX64))
+  {
+    # x64 programs actually install under "Program Files" on arm64,
+    # but this would conflict with the native installation.
+    $ProgramFilesName = "Program Files (x64)"
```
Comment (compnerd) [inline build.ps1:707]: Interesting, how does Microsoft handle the conflict?  What if you need/want both versions?

Comment (compnerd) [inline build.ps1:707]:
```suggestion
    $ProgramFilesName = "Program Files (Amd64)"
```

This would match `Program Files (Arm)` better ...

File: build.ps1 [hunk 2]
```diff
@@ -695,8 +692,30 @@ function Consolidate-RedistInstall($Arch)
...
+  }
+  else
+  {
+    # arm64 cannot be installed on x64
```
Comment (compnerd) [inline build.ps1:711]: 😢

File: build.ps1 [hunk 3]
```diff
@@ -695,8 +692,30 @@ function Consolidate-RedistInstall($Arch)
...
+    return
+  }
+
+  $RedistInstallRoot = "S:\$ProgramFilesName\swift\runtime-development"
```
Comment (compnerd) [inline build.ps1:715]: I wonder if we want to use a parameter for the stage location, but that is outside the scope of this change I think.

File: build.ps1 [hunk 4]
```diff
@@ -695,8 +692,30 @@ function Consolidate-RedistInstall($Arch)
...
+
+  Remove-Item -Force -Recurse $RedistInstallRoot -ErrorAction Ignore
+  Copy-Dir "$($HostArch.SDKInstallRoot)\usr\bin" "$RedistInstallRoot\usr"
```
Comment (compnerd) [inline build.ps1:718]: This should be `$($Arch.SDKInstallRoot)` no?

**Review by compnerd — APPROVED** [review]

File: build.ps1 (hunk 1 above)
Comment (tristanlabelle) [reply build.ps1:707]: I haven't found anything official but it seems like this is not a scenario that they went through extra lengths to support.

https://stackoverflow.com/questions/74402454/what-is-each-program-files-folder-purpose-in-arm64-architecture
